lib/ClusterProvider.py

```diff
@@ -0,0 +1,46 @@
+import common
+import os
+from robot.api import logger
+from robot.libraries.BuiltIn import BuiltIn
+
+
+ROBOT_LIBRARY_SCOPE = 'SUITE'
+AUTH_COMMAND = ''
+
+
+def auth_wrap(cmd):
+    return AUTH_COMMAND+' && '+cmd
+
+
+class ClusterProvider(common.CommandRunner):
+    def __init__(self):
+        self.provider = os.getenv("CLUSTER_PROVIDER", default='kind')
+        self.cluster_name = ''
+
+    def get_current_version(self):
+        return BuiltIn().get_variable_value('${version}')
+
+    def create_test_cluster_with_kubernetes_version(self, kube_version):
+        global AUTH_COMMAND
+        self.cluster_name = f'helm-acceptance-test-{kube_version}'
+        self.call_cluster_provisioner_function(
+            'setup_cluster')
+        AUTH_COMMAND = self.call_cluster_provisioner_function(
+            'get_cluster_auth')
+
+    def wait_for_cluster(self):
+        return self.call_cluster_provisioner_function('wait_for_cluster')
+
+    def delete_test_cluster(self):
+        return self.call_cluster_provisioner_function('delete_cluster')
+
+    def cleanup_all_test_clusters(self):
+        return self.call_cluster_provisioner_function('cleanup_all_test_clusters')
+
+    def get_cluster_auth(self):
+        return self.call_cluster_provisioner_function('get_cluster_auth')
+
+    def call_cluster_provisioner_function(self, func, args=''):
+        c = f'{self.provider}_{func} {self.cluster_name} {self.get_current_version()} {args}'
+        self.run_command(c)
```

Comment:
I believe the `self.run_command(c)` need to handle working directory or relative paths properly...
While running the tests not directly via `robot kubernetes_versions.robot` but using the `make acceptance` (which is the way how the tests would ultimately be executed) I get errors complaining that `command not found`:

For example by running:
```
CLUSTER_VERSION=1.14.6,1.15.3 CLUSTER_PROVIDER=kind ROBOT_RUN_TESTS=kubernetes_versions.robot ROBOT_HELM_V3=1 ROBOT_DEBUG_LEVEL=3 ROBOT_HELM_PATH=/usr/bin make acceptance
```

The test fails and in the logs I see the following:
```
KEYWORD ClusterProvider . Create Test Cluster With Kubernetes Version ${kube_version}

+ kind_setup_cluster helm-acceptance-test-1.14.6 1.14.6 
/bin/bash: kind_setup_cluster: command not found
+ kind_get_cluster_auth helm-acceptance-test-1.14.6 1.14.6 
/bin/bash: kind_get_cluster_auth: command not found
--
```

Reply:
Oh, I see the `PATH` needs to include the scripts directory... so that should go into the `scripts/acceptance.sh`:
```patch
diff --git a/scripts/acceptance.sh b/scripts/acceptance.sh
index 733d575..7b574e6 100755
--- a/scripts/acceptance.sh
+++ b/scripts/acceptance.sh
@@ -98,7 +98,7 @@ SUITES_TO_RUN=${SUITES_TO_RUN:-testsuites}
 if [ ! -z "${ROBOT_HELM_PATH}" ]; then
    export PATH="${ROBOT_HELM_PATH}:${PATH}"
 fi
-export PATH="${ROBOT_VENV_DIR}/bin:${PATH}"
+export PATH="${ROBOT_VENV_DIR}/bin:${PATH}:scripts/cluster_providers"
 
 export XDG_CACHE_HOME=${TMP_DIR}/cache && mkdir -p ${XDG_CACHE_HOME}
 export XDG_CONFIG_HOME=${TMP_DIR}/config && mkdir -p ${XDG_CONFIG_HOME}
```